### PR TITLE
Fixed #26162 -- Checked query name clashes of hidden relationships.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

File: django/db/models/fields/related.py [hunk 1]
```diff
@@ -211,12 +205,14 @@ def _check_clashes(self):
         #         foreign = models.ForeignKey(Target)
         #         m2m = models.ManyToManyField(Target)
 
-        rel_opts = self.remote_field.model._meta
         # rel_opts.object_name == "Target"
+        rel_opts = self.remote_field.model._meta
+        # If the field doesn't install backward relation on the target model (so
+        # `is_hidden` returns True), then there are no clashes to check and we
```
Comment (Ian-Foote) [inline django/db/models/fields/related.py]: This comment has lost its end: `# can skip these fields.`

Comment (charettes) [inline django/db/models/fields/related.py]: Thanks!

**Comment by charettes** [comment]
@Ian-Foote can you confirm you now get a check failure?

**Comment by Ian-Foote** [comment]
Yes, I do.

File: django/db/models/fields/related.py [hunk 2]
```diff
@@ -211,12 +205,15 @@ def _check_clashes(self):
         #         foreign = models.ForeignKey(Target)
         #         m2m = models.ManyToManyField(Target)
 
-        rel_opts = self.remote_field.model._meta
         # rel_opts.object_name == "Target"
+        rel_opts = self.remote_field.model._meta
+        # If the field doesn't install backward relation on the target model (so
```
Comment (timgraham) [inline django/db/models/fields/related.py]: a backward relation

File: docs/releases/1.8.10.txt [hunk 3]
```diff
@@ -9,4 +9,4 @@ Django 1.8.10 fixes several bugs in 1.8.9.
 Bugfixes
 ========
 
-* ...
+* Checked query name clashes of hidden relationships (:ticket:`26162`).
```
Comment (timgraham) [inline docs/releases/1.8.10.txt]: To make it more obvious where the checking happens, I'd say, "Added system checks for query name clashes..."

**Pull request merged**